dyson/expressions/ccsd.py

```diff
@@ -16,28 +16,29 @@ class CCSD_1h(BaseExpression):
 
     hermitian = False
 
-    def __init__(self, *args, t1=None, t2=None, l1=None, l2=None, **kwargs):
+    def __init__(self, *args, ccsd=None, t1=None, t2=None, l1=None, l2=None, **kwargs):
         BaseExpression.__init__(self, *args, **kwargs)
 
-        ccsd = cc.CCSD(self.mf, mo_coeff=self.mo_coeff, mo_occ=self.mo_occ)
-        ccsd.verbose = 0
-
-        if t1 is None or l1 is None:
+        if ccsd is None:
+            ccsd = cc.CCSD(self.mf, mo_coeff=self.mo_coeff, mo_occ=self.mo_occ)
+            ccsd.verbose = 0
+        
+        # Update amplitudes if provided as kwargs
+        ccsd.t1 = t1 if t1 is not None else ccsd.t1
+        ccsd.t2 = t2 if t2 is not None else ccsd.t2 
+        ccsd.l1 = l1 if l1 is not None else ccsd.l1
+        ccsd.l2 = l2 if l2 is not None else ccsd.l2
```

Comment:
we should avoid changing the `ccsd` object in-place -- this may have undesired behaviour for the user. Can you refactor this to just change `self.t1` etc. instead?

Reply:
Sure